lib/index.js

```diff
@@ -27,11 +28,13 @@ function plugin(options){
       if (typeof data.excerpt === 'string' && data.excerpt.length) {
         return; // don't mutate data
       }
-
-      debug('storing excerpt: %s', file);
       var $ = cheerio.load(data.contents.toString());
       var p = $('p').first();
-      data.excerpt = $.html(p).trim();
+      
+      data.excerpt = options && options.selector==='innerHTML'? $(p).html():$.html(p) 
+      data.excerpt && data.excerpt.trim();
```

Comment:
I'm pretty sure this line doesn't actually do anything. String#trim doesn't mutate the string, it just returns a new string with the outer whitespace removed. Strings in JS are immutable.